### PR TITLE
create ClientManager class for managing clients and connection sessio…

### PR DESCRIPTION
**Pull request by Lucikr**
…ns, refactor code to work with new functionality

### CODE REVIEW

File: pgmanage/app/client_manager/client_manager.py [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+        tab = self.get_tab(client_id=client_id, tab_id=tab_id, conn_tab_id=conn_tab_id)
+        if tab is None:
+            return
+
+        if tab_id:
+            del client["connection_sessions"][conn_tab_id]["tab_list"][tab_id]
+        else:
+            del client["connection_sessions"][conn_tab_id]
+
+        if (
```
Comment (eug3nix) [inline pgmanage/app/client_manager/client_manager.py:267]: maybe replace with smth like
if tab["type"] in ["query", "console", "connection", "edit"] ?

File: pgmanage/app/client_manager/client_manager.py [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+                tab["omnidatabase"].v_connection.Cancel(False)
+            except Exception:
+                pass
+            try:
+                tab["omnidatabase"].v_connection.Close()
+            except Exception:
+                pass
+
+        elif tab["type"] == "debug":
+            tab["cancelled"] = True
```
Comment (eug3nix) [inline pgmanage/app/client_manager/client_manager.py:283]: should this part of code be extracted into a separate method?

File: pgmanage/app/client_manager/client_manager.py [hunk 3]
```diff
@@ -0,0 +1,544 @@
...
+                tab["omnidatabase_control"].v_connection.Close()
+            except Exception:
+                pass
+            try:
+                tab["omnidatabase_debug"].v_connection.Close()
+            except Exception:
+                pass
+
+        elif tab["type"] == "terminal":
+            if tab["thread"] is not None:
```
Comment (eug3nix) [inline pgmanage/app/client_manager/client_manager.py:302]: should this part of code be extracted into a separate method?

File: pgmanage/app/client_manager/client_manager.py (hunk 1 above)
Comment (Lucikr) [reply pgmanage/app/client_manager/client_manager.py]: Done.

File: pgmanage/app/client_manager/client_manager.py (hunk 2 above)
Comment (Lucikr) [reply pgmanage/app/client_manager/client_manager.py]: Yes.
I've created method '_close_debug_tab' to extract this part of the code into a separate method.

File: pgmanage/app/client_manager/client_manager.py (hunk 3 above)
Comment (Lucikr) [reply pgmanage/app/client_manager/client_manager.py]: Yes.
I've created method '_close_terminal_tab' to extract this part of the code into a separate method.

**Pull request merged**